docs/api/internal.json

```diff
@@ -130,6 +62,90 @@
       }
     },
     "/wp-json/wpjm-internal/v1/promoted-jobs/{id}": {
+      "get": {
+        "tags": ["promoted-jobs"],
+        "summary": "Retrieve job data for a job that will be promoted",
+        "operationId": "get-promotedjob",
+        "parameters": [
+          {
+            "name": "id",
+            "in": "path",
+            "description": "ID of the job to retrieve",
+            "required": true,
+            "schema": {
+              "type": "integer"
+            }
+          },
+          {
+            "name": "user_id",
+            "in": "query",
+            "description": "ID of the user making the request",
+            "required": true,
+            "schema": {
+              "type": "integer"
+            }
+          },
+          {
+            "name": "token",
+            "in": "query",
+            "description": "Random token encoded generated by this plugin",
+            "required": true,
+            "schema": {
+              "type": "string"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful response",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "oneOf": [
+                    {
+                      "type": "object",
+                      "description": "The request is completely valid and the job data is returned",
+                      "properties": {
+                        "verified": {
+                          "type": "boolean",
+                          "enum": [
+                            true
+                          ]
+                        },
+                        "job_data": {
+                          "$ref": "#/components/schemas/JobData"
+                        }
+                      }
+                    },
+                    {
+                      "type": "object",
+                      "description": "The token/user ID is expired or invalid",
```

Comment:
Shouldn't it be returned as `403` instead of `200`?

Reply:
My idea of keeping it as `200` is to "reply" to the client that the answer is "valid", but not verified.

For instance, depending on the settings, the server could already reply with 403 (if the user has some plugin to enforce API authentication for some reason, for instance). By replying with `200`, we can kinda pass the message that the request itself is OK (so it is not a server configuration error), BUT the token isn't verified.

Does that make sense?

Reply:
Yep! Sounds good! ;)